core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -104,9 +104,15 @@ oppia.controller('CollectionPlayer', [
     };
 
     $scope.getNonRecommendedCollectionNodes = function() {
-      var displayedExplorationIds = (
-        [$scope.collectionPlaythrough.getNextExplorationId()].concat(
-          $scope.collectionPlaythrough.getCompletedExplorationIds()));
+      var displayedExplorationIds = [];
+      if ($scope.collectionPlaythrough.getNextExplorationId() !== null) {
+        displayedExplorationIds = (
```

Comment:
You can simplify this -- set displayedExplorationIds as the thing that's common to both branches and then have a single 'if' that checks if there's a next exp ID; if so, add it to the list.

In general, try to have as little branching logic as possible, to reduce code complexity.